greynoise/actions/workflows/ip_lookup.yaml

```diff
@@ -0,0 +1,58 @@
+---
+
+version: 1.0
+
+description: >-
+  A multi-lookup workflow to provide RIOT or Context data for known IPs in GreyNoise.
+
+input:
+  - ip
+
+vars:
+  - is_riot: false
+  - is_seen: false
+  - result: ""
+
+output:
+  - is_riot: <% ctx().is_riot %>
+  - is_seen: <% ctx().is_seen %>
+  - result: <% ctx().result %>
+
+tasks:
+  check_riot:
+    action: greynoise.riot_ip
+    input:
+      ip: <% ctx().ip %>
+    next:
+      - when: <% succeeded() and result().result.riot = true %>
+        publish:
+          result=<% result().result %>
+          is_riot=true
+      - when: <% succeeded() and result().result.riot = false %>
+        do: check_quick
+      - when: <% failed() and not result().result %>
+        publish:
+          result=<% result().stderr %>
+      - when: <% failed() and result() %>
+        publish:
+          result=<% result().result %>
+  check_quick:
+    action: greynoise.quick_ip
+    input:
+      ip: <% ctx().ip %>
+    next:
+      - when: <% succeeded() and result().result[0].noise = true %>
```

Comment:
If the action successfully returns but `result().result` is empty, this will cause a YAQL error, which is more difficult to debug. I would add a length check in this.

Reply:
added length check.